src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -195,7 +202,12 @@ public CourseResource(Environment env, UserService userService, CourseService co
                     return ResponseEntity.badRequest().headers(HeaderUtil.createAlert(applicationName, "The shortname is invalid", "shortnameInvalid")).body(null);
                 }
                 checkIfGroupsExists(updatedCourse);
+                final var oldInstructorGroup = existingCourse.get().getInstructorGroupName();
```

Comment:
please add some comments what this code wants to achieve. It's not really clear to me at the moment why we need this here

Reply:
I added some comments.
Basically, we need to calculate the diff between the old instructors/tutors and the new ones. Based on that, the `VCSUserManagementService` can then remove/add/update users for all exercise repositories in the related course.